usvm-ts/src/main/kotlin/org/usvm/machine/state/TSState.kt

```diff
@@ -26,6 +27,7 @@ class TSState(
     forkPoints: PathNode<PathNode<EtsStmt>> = PathNode.root(),
     var methodResult: TSMethodResult = TSMethodResult.NoCall,
     targets: UTargetsSet<TSTarget, EtsStmt> = UTargetsSet.empty(),
+    private val localToSort: MutableMap<EtsMethod, MutableMap<Int, USort>> = hashMapOf(),
```

Comment:
Maybe it is better to use persistent map?

Reply:
I think it's ok for now due to multiple potential put actions during getOrPutSoertForLocal methods

Reply:
You can use `UPersistentHashMap` which is designed to perform write operations. You can find usage examples in `UMemory` (`regions` property)